File: 刷题/golang/1616. 分割两个字符串得到回文串.go
func checkPalindrome(a string, left int, right int) bool {
    for left<right {
        if a[left] != a[right] {
            return false
        }
        left++
        right--
    }
    return true
}
func chekcAB(a, b string) bool {
    left, right := 0, len(a)-1
    for left < right && a[left] == b[right] {
        left++
        right--
    }
    if left >= right {
        return true
    } 
    return checkPalindrome(a, left, right) || checkPalindrome(b, left, right)
}
func checkPalindromeFormation(a string, b string) bool {
    return chekcAB(a, b) || chekcAB(b, a)
}

File: 刷题/cpp/1616. 分割两个字符串得到回文串.cpp
class Solution {
public:
    bool check(string x){
        int n=x.length();
        for(int i=0;i<=n/2;i++){
            if(x[i]!=x[n-i-1]) return false;
        }
        return true;
    }
    bool checkAB(const string &a,const string &b){
        int n=a.size();
        int left=0, right=n-1;
        while(left<right && a[left]==b[right]){
            left++;
            right--;
        }
        if(left>=right){
            return true;
        }
        return check(a.substr(left,right-left+1)) || check(b.substr(left,right-left+1));
    }
    bool checkPalindromeFormation(string a, string b) {
        return checkAB(a,b) || checkAB(b,a);
    }
};